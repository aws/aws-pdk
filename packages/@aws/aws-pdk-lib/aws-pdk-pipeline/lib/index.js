"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
__exportStar(require("./pdk-pipeline"), exports);
__exportStar(require("./pdk-pipeline-ts-project"), exports);
__exportStar(require("./pdk-pipeline-py-project"), exports);
__exportStar(require("./pdk-pipeline-java-project"), exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi9zcmMvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7Ozs7O0FBQUEsaURBQStCO0FBQy9CLDREQUEwQztBQUMxQyw0REFBMEM7QUFDMUMsOERBQTRDIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSBcIi4vcGRrLXBpcGVsaW5lXCI7XG5leHBvcnQgKiBmcm9tIFwiLi9wZGstcGlwZWxpbmUtdHMtcHJvamVjdFwiO1xuZXhwb3J0ICogZnJvbSBcIi4vcGRrLXBpcGVsaW5lLXB5LXByb2plY3RcIjtcbmV4cG9ydCAqIGZyb20gXCIuL3Bkay1waXBlbGluZS1qYXZhLXByb2plY3RcIjtcbiJdfQ==