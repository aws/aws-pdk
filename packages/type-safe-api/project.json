{
  "name": "@aws/type-safe-api",
  "root": "packages/type-safe-api",
  "targets": {
    "build": {
      "inputs": [
        "default",
        "^default",
        "!{projectRoot}/.jsii",
        "!{projectRoot}/lib/**/*",
        "!{projectRoot}/dist/**/*",
        "!{projectRoot}/coverage/**/*",
        "!{projectRoot}/test-reports/**/*",
        "!{projectRoot}/docs/api/**/*"
      ],
      "outputs": [
        "{projectRoot}/.jsii",
        "{projectRoot}/lib",
        "{projectRoot}/dist",
        "{projectRoot}/coverage",
        "{projectRoot}/test-reports",
        "{projectRoot}/docs/api"
      ],
      "dependsOn": [
        "^build"
      ],
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen build",
        "cwd": "packages/type-safe-api"
      }
    },
    "default": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen default",
        "cwd": "packages/type-safe-api"
      }
    },
    "pre-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen pre-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "post-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen post-compile",
        "cwd": "packages/type-safe-api"
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test",
        "cwd": "packages/type-safe-api"
      }
    },
    "package": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package",
        "cwd": "packages/type-safe-api"
      }
    },
    "test:watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test:watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen watch",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint",
        "cwd": "packages/type-safe-api"
      }
    },
    "compat": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compat",
        "cwd": "packages/type-safe-api"
      }
    },
    "package-all": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package-all",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:java": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:java",
        "cwd": "packages/type-safe-api"
      }
    },
    "package:python": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:python",
        "cwd": "packages/type-safe-api"
      }
    },
    "eslint-staged": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint-staged",
        "cwd": "packages/type-safe-api"
      }
    },
    "jest": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen jest",
        "cwd": "packages/type-safe-api"
      }
    },
    "docgen": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen docgen",
        "cwd": "packages/type-safe-api"
      }
    }
  },
  "implicitDependencies": [
    "smithy-async-transformer"
  ],
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"pnpm exec projen\"."
}
