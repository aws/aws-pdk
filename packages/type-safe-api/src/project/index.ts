/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./model/type-safe-api-model-build";
export * from "./model/type-safe-api-async-model-build";
export * from "./model/smithy/smithy-project-definition";
export * from "./model/openapi/open-api-project-definition";
export * from "./type-safe-api-project";
export * from "./model/type-safe-api-model-project";
export * from "./model/smithy/smithy-definition";
export * from "./model/openapi/open-api-definition";
export * from "./type-safe-websocket-api-project";
export * from "./model/type-safe-websocket-api-model-project";
export * from "./model/smithy/smithy-async-definition";
export * from "./model/openapi/open-api-async-definition";
export * from "./model/smithy/types";
export * from "./model/openapi/open-api-model-project";
export * from "./model/openapi/open-api-async-model-project";
export * from "./model/smithy/smithy-async-model-project";
export * from "./model/smithy/smithy-model-project";
export * from "./languages";
export * from "./types";
export * from "./typescript-project-options";
export * from "./python-project-options";
export * from "./java-project-options";
