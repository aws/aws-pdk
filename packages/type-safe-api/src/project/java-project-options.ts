// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
import { github, GitOptions, IgnoreFileOptions, java, LoggerOptions, Project, ProjectType, ProjenrcJsonOptions, RenovatebotOptions, SampleReadmeProps } from 'projen';

/**
 * JavaProjectOptions
 */
export interface JavaProjectOptions {
  /**
   * The java package to use for the code sample.
   * @default "org.acme"
   * @stability experimental
   */
  readonly sampleJavaPackage?: string;
  /**
   * Include sample code and test if the relevant directories don't exist.
   * @default true
   * @stability experimental
   */
  readonly sample?: boolean;
  /**
   * List of test dependencies for this project.
   * Dependencies use the format: `<groupId>/<artifactId>@<semver>`
   *
   * Additional dependencies can be added via `project.addTestDependency()`.
   * @default []
   * @stability experimental
   * @featured true
   */
  readonly testDeps?: Array<string>;
  /**
   * Options related to projenrc in java.
   * @default - default options
   * @stability experimental
   */
  readonly projenrcJavaOptions?: java.ProjenrcOptions;
  /**
   * Use projenrc in java.
   * This will install `projen` as a java dependency and will add a `synth` task which
   * will compile & execute `main()` from `src/main/java/projenrc.java`.
   * @default true
   * @stability experimental
   */
  readonly projenrcJava?: boolean;
  /**
   * Packaging options.
   * @default - defaults
   * @stability experimental
   */
  readonly packagingOptions?: java.MavenPackagingOptions;
  /**
   * junit options.
   * @default - defaults
   * @stability experimental
   */
  readonly junitOptions?: java.JunitOptions;
  /**
   * Include junit tests.
   * @default true
   * @stability experimental
   */
  readonly junit?: boolean;
  /**
   * Final artifact output directory.
   * @default "dist/java"
   * @stability experimental
   */
  readonly distdir?: string;
  /**
   * List of runtime dependencies for this project.
   * Dependencies use the format: `<groupId>/<artifactId>@<semver>`
   *
   * Additional dependencies can be added via `project.addDependency()`.
   * @default []
   * @stability experimental
   * @featured true
   */
  readonly deps?: Array<string>;
  /**
   * Compile options.
   * @default - defaults
   * @stability experimental
   */
  readonly compileOptions?: java.MavenCompileOptions;
  /**
   * The URL, like the name, is not required.
   * This is a nice gesture for
   * projects users, however, so that they know where the project lives.
   * @default undefined
   * @stability experimental
   * @featured true
   */
  readonly url?: string;
  /**
   * Project packaging format.
   * @default "jar"
   * @stability experimental
   */
  readonly packaging?: string;
  /**
   * Description of a project is always good.
   * Although this should not replace
   * formal documentation, a quick comment to any readers of the POM is always
   * helpful.
   * @default undefined
   * @stability experimental
   * @featured true
   */
  readonly description?: string;
  /**
   * This is the last piece of the naming puzzle.
   * groupId:artifactId denotes a
   * single project but they cannot delineate which incarnation of that project
   * we are talking about. Do we want the junit:junit of 2018 (version 4.12), or
   * of 2007 (version 3.8.2)? In short: code changes, those changes should be
   * versioned, and this element keeps those versions in line. It is also used
   * within an artifact's repository to separate versions from each other.
   * my-project version 1.0 files live in the directory structure
   * $M2_REPO/org/codehaus/mojo/my-project/1.0.
   * @default "0.1.0"
   * @stability experimental
   * @featured true
   */
  readonly version?: string;
  /**
   * This is generally unique amongst an organization or a project.
   * For example,
   * all core Maven artifacts do (well, should) live under the groupId
   * org.apache.maven. Group ID's do not necessarily use the dot notation, for
   * example, the junit project. Note that the dot-notated groupId does not have
   * to correspond to the package structure that the project contains. It is,
   * however, a good practice to follow. When stored within a repository, the
   * group acts much like the Java packaging structure does in an operating
   * system. The dots are replaced by OS specific directory separators (such as
   * '/' in Unix) which becomes a relative directory structure from the base
   * repository. In the example given, the org.codehaus.mojo group lives within
   * the directory $M2_REPO/org/codehaus/mojo.
   * @default "org.acme"
   * @stability experimental
   * @featured true
   */
  readonly groupId?: string;
  /**
   * The artifactId is generally the name that the project is known by.
   * Although
   * the groupId is important, people within the group will rarely mention the
   * groupId in discussion (they are often all be the same ID, such as the
   * MojoHaus project groupId: org.codehaus.mojo). It, along with the groupId,
   * creates a key that separates this project from every other project in the
   * world (at least, it should :) ). Along with the groupId, the artifactId
   * fully defines the artifact's living quarters within the repository. In the
   * case of the above project, my-project lives in
   * $M2_REPO/org/codehaus/mojo/my-project.
   * @default "my-app"
   * @stability experimental
   * @featured true
   */
  readonly artifactId?: string;
  /**
   * Enable VSCode integration.
   * Enabled by default for root projects. Disabled for non-root projects.
   * @default true
   * @stability experimental
   */
  readonly vscode?: boolean;
  /**
   * Auto-close stale issues and pull requests.
   * To disable set `stale` to `false`.
   * @default - see defaults in `StaleOptions`
   * @stability experimental
   */
  readonly staleOptions?: github.StaleOptions;
  /**
   * Auto-close of stale issues and pull request.
   * See `staleOptions` for options.
   * @default false
   * @stability experimental
   */
  readonly stale?: boolean;
  /**
   * The README setup.
   * @default - { filename: 'README.md', contents: '# replace this' }
   * @stability experimental
   */
  readonly readme?: SampleReadmeProps;
  /**
   * The name of a secret which includes a GitHub Personal Access Token to be used by projen workflows.
   * This token needs to have the `repo`, `workflows`
   * and `packages` scope.
   * @default "PROJEN_GITHUB_TOKEN"
   * @deprecated use `projenCredentials`
   * @stability deprecated
   */
  readonly projenTokenSecret?: string;
  /**
   * Choose a method of providing GitHub API access for projen workflows.
   * @default - use a personal access token named PROJEN_GITHUB_TOKEN
   * @stability experimental
   */
  readonly projenCredentials?: github.GithubCredentials;
  /**
   * Which type of project this is (library/app).
   * @default ProjectType.UNKNOWN
   * @deprecated no longer supported at the base project level
   * @stability deprecated
   */
  readonly projectType?: ProjectType;
  /**
   * Options for mergify.
   * @default - default options
   * @deprecated use `githubOptions.mergifyOptions` instead
   * @stability deprecated
   */
  readonly mergifyOptions?: github.MergifyOptions;
  /**
   * Whether mergify should be enabled on this repository or not.
   * @default true
   * @deprecated use `githubOptions.mergify` instead
   * @stability deprecated
   */
  readonly mergify?: boolean;
  /**
   * Add a Gitpod development environment.
   * @default false
   * @stability experimental
   */
  readonly gitpod?: boolean;
  /**
   * Options for GitHub integration.
   * @default - see GitHubOptions
   * @stability experimental
   */
  readonly githubOptions?: github.GitHubOptions;
  /**
   * Enable GitHub integration.
   * Enabled by default for root projects. Disabled for non-root projects.
   * @default true
   * @stability experimental
   */
  readonly github?: boolean;
  /**
   * Add a VSCode development environment (used for GitHub Codespaces).
   * @default false
   * @stability experimental
   */
  readonly devContainer?: boolean;
  /**
   * Add a `clobber` task which resets the repo to origin.
   * @default - true, but false for subprojects
   * @stability experimental
   */
  readonly clobber?: boolean;
  /**
   * Configure options for automatic merging on GitHub.
   * Has no effect if
   * `github.mergify` or `autoMerge` is set to false.
   * @default - see defaults in `AutoMergeOptions`
   * @stability experimental
   */
  readonly autoMergeOptions?: github.AutoMergeOptions;
  /**
   * Enable automatic merging on GitHub.
   * Has no effect if `github.mergify`
   * is set to false.
   * @default true
   * @stability experimental
   */
  readonly autoMerge?: boolean;
  /**
   * Enable and configure the 'auto approve' workflow.
   * @default - auto approve is disabled
   * @stability experimental
   */
  readonly autoApproveOptions?: github.AutoApproveOptions;
  /**
   * Options for renovatebot.
   * @default - default options
   * @stability experimental
   */
  readonly renovatebotOptions?: RenovatebotOptions;
  /**
   * Use renovatebot to handle dependency upgrades.
   * @default false
   * @stability experimental
   */
  readonly renovatebot?: boolean;
  /**
   * Options for .projenrc.json.
   * @default - default options
   * @stability experimental
   */
  readonly projenrcJsonOptions?: ProjenrcJsonOptions;
  /**
   * Generate (once) .projenrc.json (in JSON). Set to `false` in order to disable .projenrc.json generation.
   * @default false
   * @stability experimental
   */
  readonly projenrcJson?: boolean;
  /**
   * The shell command to use in order to run the projen CLI.
   * Can be used to customize in special environments.
   * @default "npx projen"
   * @stability experimental
   */
  readonly projenCommand?: string;
  /**
   * The parent project, if this project is part of a bigger project.
   * @stability experimental
   */
  readonly parent?: Project;
  /**
   * The root directory of the project.
   * Relative to this directory, all files are synthesized.
   *
   * If this project has a parent, this directory is relative to the parent
   * directory and it cannot be the same as the parent or any of it's other
   * sub-projects.
   * @default "."
   * @stability experimental
   */
  readonly outdir?: string;
  /**
   * Configure logging options such as verbosity.
   * @default {}
   * @stability experimental
   */
  readonly logging?: LoggerOptions;
  /**
   * Configuration options for git.
   * @stability experimental
   */
  readonly gitOptions?: GitOptions;
  /**
   * Configuration options for .gitignore file.
   * @stability experimental
   */
  readonly gitIgnoreOptions?: IgnoreFileOptions;
  /**
   * Whether to commit the managed files by default.
   * @default true
   * @stability experimental
   */
  readonly commitGenerated?: boolean;
  /**
   * This is the name of your project.
   * @default $BASEDIR
   * @stability experimental
   * @featured true
   */
  readonly name?: string;
}
