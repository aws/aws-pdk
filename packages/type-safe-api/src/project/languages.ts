/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
/**
 * Supported languages for runtimes and infrastructure
 */
export enum Language {
  TYPESCRIPT = "typescript",
  PYTHON = "python",
  JAVA = "java",
}

/**
 * Supported libraries for code generation
 */
export enum Library {
  /**
   * React query v4 hooks library
   * @deprecated Use react query v5 hooks library instead. This will be removed in PDK 1.0 release.
   */
  TYPESCRIPT_REACT_QUERY_HOOKS = "typescript-react-query-hooks",
  /**
   * React query v5 hooks library
   */
  TYPESCRIPT_REACT_QUERY_V5_HOOKS = "typescript-react-query-v5-hooks",
}

/**
 * Formats for documentation generation
 */
export enum DocumentationFormat {
  /**
   * HTML Documentation generated by redoc
   * @see https://github.com/Redocly/redoc
   */
  HTML_REDOC = "html_redoc",
  /**
   * OpenAPI Generator 'html2' documentation
   * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/html2.md
   */
  HTML2 = "html2",
  /**
   * OpenAPI Generator 'markdown' documentation
   * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/markdown.md
   */
  MARKDOWN = "markdown",
  /**
   * OpenAPI Generator 'plantuml' documentation
   * @see https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/plantuml.md
   */
  PLANTUML = "plantuml",
}
