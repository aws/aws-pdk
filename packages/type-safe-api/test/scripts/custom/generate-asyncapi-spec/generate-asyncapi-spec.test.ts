/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import os from "os";
import path from "path";
import { exec } from "projen/lib/util";
import { withTmpDirSnapshot } from "../../../project/snapshot-utils";

describe("Generate AsyncAPI Spec Unit Tests", () => {
  it.each(["single.yaml", "inline-body.yaml"])(
    "Generates AsyncAPI Spec for %s",
    (specFile) => {
      expect(
        withTmpDirSnapshot(os.tmpdir(), (tmpDir) => {
          const specPath = `../../../resources/specs/async/${specFile}`;
          const outputPath = path.join(
            path.relative(path.resolve(__dirname), tmpDir),
            ".asyncapi.json"
          );
          const command = `../../../../scripts/type-safe-api/custom/generate-asyncapi-spec/generate-asyncapi-spec --spec-path ${specPath} --output-path ${outputPath}`;
          exec(command, {
            cwd: path.resolve(__dirname),
          });
        })
      ).toMatchSnapshot();
    }
  );
});
