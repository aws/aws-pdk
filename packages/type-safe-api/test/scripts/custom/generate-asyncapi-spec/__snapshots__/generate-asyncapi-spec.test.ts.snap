// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate AsyncAPI Spec Unit Tests Generates AsyncAPI Spec for inline-body.yaml 1`] = `
{
  ".asyncapi.json": {
    "asyncapi": "3.0.0",
    "channels": {
      "default": {
        "address": "/",
        "messages": {
          "Java": {
            "$ref": "#/components/messages/Java",
          },
          "Python": {
            "$ref": "#/components/messages/Python",
          },
          "TypeScript": {
            "$ref": "#/components/messages/TypeScript",
          },
        },
      },
    },
    "components": {
      "messages": {
        "Java": {
          "contentType": "application/json",
          "name": "Java",
          "payload": {
            "properties": {
              "payload": {
                "properties": {
                  "field1": {
                    "type": "string",
                  },
                  "field2": {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "route": {
                "const": "Java",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "Java",
        },
        "Python": {
          "contentType": "application/json",
          "name": "Python",
          "payload": {
            "properties": {
              "payload": {
                "properties": {
                  "field1": {
                    "type": "string",
                  },
                  "field2": {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "route": {
                "const": "Python",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "Python",
        },
        "TypeScript": {
          "contentType": "application/json",
          "name": "TypeScript",
          "payload": {
            "properties": {
              "payload": {
                "properties": {
                  "field1": {
                    "type": "string",
                  },
                  "field2": {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "route": {
                "const": "TypeScript",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "TypeScript",
        },
      },
      "schemas": {},
    },
    "info": {
      "title": "Test",
      "version": "1.0.0",
    },
    "operations": {
      "Java": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/Java",
          },
        ],
        "title": "Java",
      },
      "Python": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/Python",
          },
        ],
        "title": "Python",
      },
      "TypeScript": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/TypeScript",
          },
        ],
        "title": "TypeScript",
      },
    },
  },
}
`;

exports[`Generate AsyncAPI Spec Unit Tests Generates AsyncAPI Spec for single.yaml 1`] = `
{
  ".asyncapi.json": {
    "asyncapi": "3.0.0",
    "channels": {
      "default": {
        "address": "/",
        "messages": {
          "any-request": {
            "$ref": "#/components/messages/any-request",
          },
          "bidirectional": {
            "$ref": "#/components/messages/bidirectional",
          },
          "client-to-server": {
            "$ref": "#/components/messages/client-to-server",
          },
          "different-media-type": {
            "$ref": "#/components/messages/different-media-type",
          },
          "empty-request": {
            "$ref": "#/components/messages/empty-request",
          },
          "map-request": {
            "$ref": "#/components/messages/map-request",
          },
          "multiple-content-types": {
            "$ref": "#/components/messages/multiple-content-types",
          },
          "server-to-client": {
            "$ref": "#/components/messages/server-to-client",
          },
          "without-operation-id": {
            "$ref": "#/components/messages/without-operation-id",
          },
        },
      },
    },
    "components": {
      "messages": {
        "any-request": {
          "contentType": "application/json",
          "name": "any-request",
          "payload": {
            "properties": {
              "payload": {},
              "route": {
                "const": "any-request",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "any-request",
        },
        "bidirectional": {
          "contentType": "application/json",
          "name": "bidirectional",
          "payload": {
            "properties": {
              "payload": {
                "$ref": "#/components/schemas/TestRequest",
              },
              "route": {
                "const": "bidirectional",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "bidirectional",
        },
        "client-to-server": {
          "contentType": "application/json",
          "name": "client-to-server",
          "payload": {
            "properties": {
              "payload": {
                "$ref": "#/components/schemas/TestRequest",
              },
              "route": {
                "const": "client-to-server",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "client-to-server",
        },
        "different-media-type": {
          "name": "different-media-type",
          "payload": {
            "properties": {
              "route": {
                "const": "different-media-type",
                "type": "string",
              },
            },
            "required": [
              "route",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "different-media-type",
        },
        "empty-request": {
          "name": "empty-request",
          "payload": {
            "properties": {
              "route": {
                "const": "empty-request",
                "type": "string",
              },
            },
            "required": [
              "route",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "empty-request",
        },
        "map-request": {
          "contentType": "application/json",
          "name": "map-request",
          "payload": {
            "properties": {
              "payload": {
                "$ref": "#/components/schemas/MapRequest",
              },
              "route": {
                "const": "map-request",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "map-request",
        },
        "multiple-content-types": {
          "contentType": "application/json",
          "name": "multiple-content-types",
          "payload": {
            "properties": {
              "payload": {
                "$ref": "#/components/schemas/TestRequest",
              },
              "route": {
                "const": "multiple-content-types",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "multiple-content-types",
        },
        "server-to-client": {
          "contentType": "application/json",
          "name": "server-to-client",
          "payload": {
            "properties": {
              "payload": {
                "$ref": "#/components/schemas/TestRequest",
              },
              "route": {
                "const": "server-to-client",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "server-to-client",
        },
        "without-operation-id": {
          "contentType": "application/json",
          "name": "without-operation-id",
          "payload": {
            "properties": {
              "payload": {
                "$ref": "#/components/schemas/TestRequest",
              },
              "route": {
                "const": "without-operation-id",
                "type": "string",
              },
            },
            "required": [
              "route",
              "payload",
            ],
            "type": "object",
          },
          "tags": [],
          "title": "without-operation-id",
        },
      },
      "schemas": {
        "ApiError": {
          "properties": {
            "errorMessage": {
              "type": "string",
            },
          },
          "required": [
            "errorMessage",
          ],
          "type": "object",
        },
        "MapRequest": {
          "properties": {
            "mapProperty": {
              "additionalProperties": {
                "properties": {
                  "a": {
                    "type": "string",
                  },
                  "b": {
                    "type": "string",
                  },
                },
                "required": [
                  "a",
                ],
                "type": "object",
              },
              "type": "object",
            },
          },
          "required": [
            "mapProperty",
          ],
          "type": "object",
        },
        "TestRequest": {
          "description": "This is a test request",
          "properties": {
            "myInput": {
              "type": "number",
            },
          },
          "type": "object",
        },
        "TestResponse": {
          "description": "This is a test response",
          "properties": {
            "messages": {
              "items": {
                "properties": {
                  "id": {
                    "type": "integer",
                  },
                  "message": {
                    "description": "This is a message",
                    "type": "string",
                  },
                },
                "required": [
                  "id",
                ],
                "type": "object",
              },
              "type": "array",
            },
          },
          "required": [
            "messages",
          ],
          "type": "object",
        },
      },
    },
    "info": {
      "title": "Example API",
      "version": "1.0.0",
    },
    "operations": {
      "Onany-request": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/any-request",
          },
        ],
        "title": "Onany-request",
      },
      "Onbidirectional": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/bidirectional",
          },
        ],
        "title": "Onbidirectional",
      },
      "Ondifferent-media-type": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/different-media-type",
          },
        ],
        "title": "Ondifferent-media-type",
      },
      "Onempty-request": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/empty-request",
          },
        ],
        "title": "Onempty-request",
      },
      "Onmap-request": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/map-request",
          },
        ],
        "title": "Onmap-request",
      },
      "Onmultiple-content-types": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/multiple-content-types",
          },
        ],
        "title": "Onmultiple-content-types",
      },
      "Onserver-to-client": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/server-to-client",
          },
        ],
        "title": "Onserver-to-client",
      },
      "Onwithout-operation-id": {
        "action": "receive",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/without-operation-id",
          },
        ],
        "title": "Onwithout-operation-id",
      },
      "any-request": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/any-request",
          },
        ],
        "title": "any-request",
      },
      "bidirectional": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/bidirectional",
          },
        ],
        "title": "bidirectional",
      },
      "client-to-server": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/client-to-server",
          },
        ],
        "title": "client-to-server",
      },
      "different-media-type": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/different-media-type",
          },
        ],
        "title": "different-media-type",
      },
      "empty-request": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/empty-request",
          },
        ],
        "title": "empty-request",
      },
      "map-request": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/map-request",
          },
        ],
        "title": "map-request",
      },
      "multiple-content-types": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/multiple-content-types",
          },
        ],
        "title": "multiple-content-types",
      },
      "without-operation-id": {
        "action": "send",
        "channel": {
          "$ref": "#/channels/default",
        },
        "messages": [
          {
            "$ref": "#/channels/default/messages/without-operation-id",
          },
        ],
        "title": "without-operation-id",
      },
    },
  },
}
`;
